components/gateway/internal/auditlog/service.go

```diff
@@ -77,60 +64,107 @@ func NewService(client AuditlogClient, msgFactory AuditlogMessageFactory) *Servi
 	return &Service{client: client, msgFactory: msgFactory}
 }
 
-func (svc *Service) Log(ctx context.Context, request, response string, claims proxy.Claims) error {
-	graphqlResponse, err := svc.parseResponse(response)
+func (svc *Service) PreLog(ctx context.Context, msg proxy.AuditlogMessage) error {
+	correlationID := msg.CorrelationIDHeaders[correlation.RequestIDHeaderKey]
+	configChangeMsg := svc.createConfigChangeMsg(msg.Claims, msg.Request)
+	configChangeMsg.Attributes = append(configChangeMsg.Attributes,
+		model.Attribute{
+			Name: "request",
+			Old:  "",
+			New:  msg.Request,
+		}, model.Attribute{
+			Name: "correlation-id",
+			Old:  "",
+			New:  correlationID,
+		})
+
+	err := svc.client.LogConfigurationChange(ctx, configChangeMsg)
+	return errors.Wrap(err, "while sending configuration pre-change")
+}
+
+func (svc *Service) Log(ctx context.Context, msg proxy.AuditlogMessage) error {
+	graphqlResponse, err := svc.parseResponse(msg.Response)
 	if err != nil {
 		return errors.Wrap(err, "while parsing response")
 	}
 
-	if len(graphqlResponse.Errors) == 0 {
-		log := svc.createConfigChangeMsg(claims, request)
-		log.Attributes = append(log.Attributes, model.Attribute{
-			Name: "response",
-			Old:  "",
-			New:  "success",
-		})
+	correlationID := msg.CorrelationIDHeaders[correlation.RequestIDHeaderKey]
 
-		err = svc.client.LogConfigurationChange(ctx, log)
+	if len(graphqlResponse.Errors) == 0 {
+		configChangeMsg := svc.createConfigChangeMsg(msg.Claims, msg.Request)
+		configChangeMsg.Attributes = append(configChangeMsg.Attributes,
+			model.Attribute{
+				Name: "request",
+				Old:  "",
+				New:  msg.Request,
+			}, model.Attribute{
+				Name: "response",
+				Old:  "",
+				New:  "success",
+			}, model.Attribute{
```

Comment:
Correlation ID seems something that will always be added to a message, so better add it to the constructor - `createConfigChangeMsg`